src/common/provider.ts

```diff
@@ -1,10 +1,43 @@
-import { JsonRpcProvider } from '@ethersproject/providers';
+import { providers, utils } from 'ethers';
 import { ConfigService } from '@nestjs/config';
 import { Injectable } from '@nestjs/common';
+import { Logger } from '../modules/logger/logger.service';
+
+const { poll } = utils;
 
 @Injectable()
-export class Provider extends JsonRpcProvider {
-  constructor(configService: ConfigService) {
+export class Provider extends providers.JsonRpcProvider {
+  private attempts: number;
+
+  constructor(configService: ConfigService, private logger: Logger) {
     super(configService.get('ENS_URL'));
+    this.attempts = configService.get('CHAIN_REQUEST_ATTEMPTS');
```

Comment:
Do you think we should have a default value for `this.attempts` if `CHAIN_REQUEST_ATTEMPTS` is not set? Otherwise, I think `if (attempts >= this.attempts)` could give us an endless loop of requests and errors

Reply:
I think that current defaul 0 attempts is reasonable default, which means that request will not be retry. Could you elaborate why this is causes endless loop? Rejected callback should not be rescheduled by `poll`. I'll try to test this
This defaul should be set explicitly though with explanation

Reply:
> Could you elaborate why this is causes endless loop?
Maybe it doesn't, I'm not sure. I think with the default value for `CHAIN_REQUEST_ATTEMPTS` though, this is not a concern